crates/miden-objects/src/transaction/inputs/mod.rs

```diff
@@ -142,14 +175,16 @@ impl Serializable for TransactionInputs {
     }
 }
 
-impl Deserializable for TransactionInputs {
+impl Deserializable for TransactionExecutionInputs {
     fn read_from<R: ByteReader>(source: &mut R) -> Result<Self, DeserializationError> {
         let partial_account = PartialAccount::read_from(source)?;
         let block_header = BlockHeader::read_from(source)?;
         let blockchain = PartialBlockchain::read_from(source)?;
+        let tx_inputs = TransactionInputs::new(partial_account, block_header, blockchain)
+            .expect("serialized tx inputs should be correct");
         let input_notes = InputNotes::read_from(source)?;
```

Comment:
If we deserialize an arbitrary block header and blockchain, they could contain anything and are not guaranteed to be consistent with each other, so I don't think we can `expect` the `TransactionInputs::new` call here. I think we need to propagate the error.